app/components/viewer/ReplayLines.tsx

```diff
@@ -28,11 +29,24 @@ export const LineTypes: { [name: string]: LineType } = {
 interface ReplayLineProps {
     replay: ReplayData;
     lineType: LineType;
+    range: number[];
     replayLineOpacity: number;
 }
-const ReplayLine = ({ replay, lineType, replayLineOpacity }: ReplayLineProps) => {
-    const points = useMemo(() => replay.samples.map((sample) => sample.position), [replay]);
-    const colorBuffer = useMemo(() => lineType.colorsCallback(replay), [replay, lineType, replay.color]);
+const ReplayLine = ({
+    replay, lineType, range, replayLineOpacity,
+}: ReplayLineProps) => {
+    const points = useMemo(() => {
+        if (range.length > 0) {
+            return replay.samples
+                .filter(
+                    (sample) => sample.currentRaceTime >= range[0] && sample.currentRaceTime <= range[1],
+                )
+                .map((sample) => sample.position);
+        }
+        return replay.samples.map((sample) => sample.position);
+    }, [replay.samples, range]);
```

Comment:
Is this range filtering really necessary? I don't see any changes in the preview build? And apart from that, I'm not quite sure whether this is something we need, might be better to keep it simple.